pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/RealtimeTableDataManager.java

```diff
@@ -226,7 +227,7 @@ public void addSegment(String segmentName, TableConfig tableConfig, IndexLoading
         _logger.error("Not adding segment {}", segmentName);
         throw new RuntimeException("Mismatching schema/table config for " + _tableNameWithType);
       }
-      VirtualColumnProviderFactory.addBuiltInVirtualColumnsToSchema(schema);
+      VirtualColumnProviderFactory.addBuiltInVirtualColumnsToSegmentSchema(schema, NetUtil.getHostnameOrAddress(), segmentName);
```

Comment:
```suggestion
      VirtualColumnProviderFactory.addBuiltInVirtualColumnsToSegmentSchema(schema, segmentName);
```
Given that this is a static function, why not let VirtualColumnProviderFactory determine the hostName (and perhaps anything else that may be needed as a part of being in cloud, for example)? It can do so on a static basis as well, instead of doing it for every segment.